novelist.py

```diff
@@ -94,6 +93,34 @@ def _confirm_same_identifier(self, metadata_objects):
         ])
         return len(novelist_ids)==1
 
+    def choose_best_metadata(self, metadata_objects, identifier):
```

Comment:
Usually in this situation I return a confidence score along with the data. (The code that calculates recursive equivalent identifiers is the closest thing in the Simplified code.) 

In this case you could return a 2-tuple (best_choice, confidence) and confidence would be the number of times the best choice showed up, divided by the total number of items.

It's up to you since your code is probably the only code that will call this method, but in situations like this I like having a rough picture of how reliable the data is.